### DISCUSSION TITLE
How to convert from unsigned to signed?

### DISCUSSION THREAD

**Discussion by joaomontenegro** [PaulRBerg/prb-math#47] [Q&A] [answered]
What's the best way of converting a unsinged fixed point uint256 (PRBMathUD60x18) into a signed fixed point uint256 (PRBMathSD59x18)?

**Comment by PaulRBerg** [comment, marked as answer]
PRBMath comes with a multitude of [casting](https://github.com/PaulRBerg/prb-math/tree/97170d0f52de76edfb5fe63c5810a8705c3b0ea7#casting-functions)and [conversion functions](https://github.com/PaulRBerg/prb-math/tree/97170d0f52de76edfb5fe63c5810a8705c3b0ea7#conversion-functions). To convert from `UD60x18` to `SD59x18`, you would do something like this:

```sol
pragma solidity >=0.8.13;

import "@prb/math/UD60x18.sol";

contract YourContract {
    function converUD60x18IntoSD59x18(UD60x18 x) external pure returns (SD59x18 result) {
        uint256 xUint = UD60x18.unwrap(x);
        require(xUint <= uint256(type(int256).max), "int256 overflow");
        int256 xInt = int256(xUint);
        result = SD59x18.wrap(xInt);
    }
}
```

I don't think that offering a cross-type convertor would be helpful, since the vast majority of PRBMath users import either the signed or the unsigned flavor (users rarely import both at the same time). In addition, for vanilla casting between `uint256` and `int256` and vice-versa, OpenZeppelin offers the [`SafeCast`](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/4e8b2fa659ba87ab916beb3eb3c8335fb1238d9f/contracts/utils/math/SafeCast.sol) library.
